app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -291,4 +291,64 @@ public function validateBeforeOrderCommit($observer) {
         $orderModel = Mage::getModel('boltpay/order');
         $orderModel->validateBeforeOrderCommit($observer);
     }
+
+    /**
+     * Automatically create an invoice after creating the shipment if createInvoiceAfterCreatingShipment setting is set to yes
+     * @param $observer
+     * @return $this
+     */
+    public function createInvoiceAfterCreatingShipment($observer)
+    {
+        $shipment = $observer->getEvent()->getShipment();
+        $order = $shipment->getOrder();
+        if (
+            !$this->boltHelper()->getExtraConfig('createInvoiceAfterCreatingShipment')
```

Comment:
@johnmccombs1, Should this be a hidden option or should this be directly advertised in the admin?  This seems like a feature that we would want to publicly promote.

Reply:
@mccottry I could. So far only one merchant has requested it, although maybe that's because we haven't advertised it. Anyone with an ERP system might benefit from it as it would mean that they could delay capture until shipment if needed. From your experience is this a common scenario that should be public?

Reply:
@johnmccombs1 
Our goal with Extra Config or "hidden configurations" is to allow ready flexibility to our system for values that we wish to discourage changing and wish to force the default behavior.  

https://app.asana.com/0/search/1145673420351635/843705092435857

Examples of this would be branding features like how the Bolt modal appears or things that affect the performance of the plugin.  For me, this is a value added feature all the way around and there seems to be no reason for it to be hidden.